File: backend/Projekatsiit2023/src/main/java/rs/ac/uns/ftn/asd/Projekatsiit2023/dto/response/ride/RideDetailsResponse.java
package rs.ac.uns.ftn.asd.Projekatsiit2023.dto.response.ride;

import lombok.Getter;
import lombok.Setter;
import rs.ac.uns.ftn.asd.Projekatsiit2023.dto.common.ReviewPresentationDTO;
import rs.ac.uns.ftn.asd.Projekatsiit2023.enums.RideStatus;
import rs.ac.uns.ftn.asd.Projekatsiit2023.enums.UserRole;
import rs.ac.uns.ftn.asd.Projekatsiit2023.models.Location;

import java.time.LocalDateTime;
import java.util.List;

@Getter
@Setter
public class RideDetailsResponse {
    // passengers
    private Long id;
    private String ordererName;
    private List<String> linkedPassengers;

    // driver
    private String driverName;
    private String vehicle;

    // route
    private Location startLocation;
    private Location endLocation;
    private List<Location> routeStops;

    // time
    private LocalDateTime startTime;
    private LocalDateTime endTime;

    // panic
    private boolean panic;
    private String panicTriggeredBy;
    private LocalDateTime panicTriggeredAt;

    // other info
    private double price;
    private RideStatus status;
    private UserRole cancelledBy;
    private List<ReviewPresentationDTO> reviews;
    private String inconsistencies;
}

File: web/taxi-app-web/src/app/models/ride-details-response.ts
export interface RideDetailsResponse {
  id: number;
  ordererName: string;

  ordererProfileImage: string | null;
  driverProfileImage: string | null;

  linkedPassengers: string[];
  driverName: string;
  vehicle: string;

  startLocation: LocationDTO;
  endLocation: LocationDTO;
  routeStops: LocationDTO[];

  startTime: string; 
  endTime: string;   

  panic: boolean;
  panicTriggeredBy: string | null;
  panicTriggeredAt: string | null;

  price: number;
  status: RideStatus;

  cancelledBy: string | null;
  reviews: ReviewDTO[];
  inconsistencies: string | null;
}

export interface LocationDTO {
  latitude: number;
  longitude: number;
  address: string;
}

export interface ReviewDTO {
    passenger: string;
    driver: string;
    driverRating: number;
    vehicleRating: number;
    comment: string;
}

export type RideStatus =
  | 'REQUESTED'
  | 'ACCEPTED'
  | 'REJECTED'
  | 'IN_PROGRESS'
  | 'CANCELLED'
  | 'FINISHED';
